src/odm2_postgres_api/routes/loggers_routes.py

```
@@ -0,0 +1,24 @@
+from fastapi import Depends, APIRouter
+from odm2_postgres_api.utils.api_pool_manager import api_pool_manager
+from odm2_postgres_api.schemas import schemas
+from typing import Optional, List
+from odm2_postgres_api.queries.core_queries import find_timeseries
+from odm2_postgres_api.queries.loggers_queries import upsert_timeseries
+
+router = APIRouter()
+
+
+@router.get("/get_timeseries_resultid/")
+async def get_timeseries_resultid(
+    data: schemas.LoggersTimeSeriesCreate,
+    connection=Depends(api_pool_manager.get_conn),
+) -> Optional[schemas.LoggersTimeSeriesId]:
```

may want to throw an error if not found, Instead of making it optional? Kinda depends on the use case

---

Guess this is a bit nitpicking, so ignore my comment freely;) I think the get_timeseries_resultid is a bit long, with the method being GET, you don't need the get in the route. Also from a user perspective I do like using the plural form where possible and return a list that is empty for no match. I also take the opportinity to complain about snakecasing routes;)

---

agreed. Routes should use dash (-) as separator instead of underscore. It is not invalid to use underscores after the domain name in urls (I think), but common practice to use dash.

---

all routes ( shared routes , begroing, mass spec) are using "_" and not "-" in this api...I feel it is too late to start changing that.